services/orchestrator/src/main/java/org/eclipse/osc/orchestrator/OrchestratorPlugin.java

```diff
@@ -0,0 +1,39 @@
+package org.eclipse.osc.orchestrator;
+
+import org.eclipse.osc.services.ocl.loader.Ocl;
+
+/**
+ * This interface describes orchestrator plugin in charge of interacting with backend fundamental APIs.
+ */
+public interface OrchestratorPlugin {
+
+    /**
+     * Register a managed service using the provided OCL descriptor.
+     *
+     * @param ocl the OCL model describing the managed service.
+     * @return the service ID on the CSP infrastructure.
+     */
+    String registerManagedService(Ocl ocl);
+
+    /**
+     * Start (exposing the managed service to the users) the managed service.
+     *
+     * @param sid the service ID to start.
+     */
+    void startManagedService(String sid);
+
+    /**
+     * Stop (hidding the managed service for the users) the managed service.
+     *
+     * @param sid the service ID to stop.
+     */
+    void stopManagedService(String sid);
+
+    /**
+     * Unregister and destroy/clean managed service resources.
+     *
+     * @param sid the service ID to unregister and destroy.
+     */
+    void unregisterManagedService(String sid);
```

Comment:
besides creating/destroying infrastructure, do we need to consider updating resources like expanding storage size, changing flavor type of vm with high performance cores for the managed service?

Reply:
Good question. Do you mean that we could have an update with a new OCL descriptor or generally speaking, starting from a OCL descriptor the CSP can change the config (or both) ?
I think it makes sense for a CSP to slightly update/change the initial OCL setup.

Reply:
the idea is coming from Terraform, it's possible to update the TF script to make it change on the Cloud side.

Reply:
It sounds good. I will update the API then. :+1:

Reply:
Agree, good point.